src/spartan/polys/multilinear.rs

```diff
@@ -287,4 +289,119 @@ mod tests {
     test_evaluation_with::<provider::bn256_grumpkin::bn256::Scalar>();
     test_evaluation_with::<provider::secp_secq::secp256k1::Scalar>();
   }
+
+  /// Returns a random ML polynomial
+  fn random<R: RngCore + CryptoRng, Scalar: PrimeField>(
+    num_vars: usize,
+    mut rng: &mut R,
+  ) -> MultilinearPolynomial<Scalar> {
+    MultilinearPolynomial::new(
+      std::iter::from_fn(|| Some(Scalar::random(&mut rng)))
+        .take(1 << num_vars)
+        .collect(),
+    )
+  }
+
+  /// This evaluates a multilinear polynomial at a partial point in the evaluation domain,
```

Comment:
Can you elaborate a bit more on the purpose of this test? What exactly are we testing here? I'm also not sure what it means to evaluate a multilinear polynomial at a "partial" point? I don't think that notion is well defined or standard.

Reply:
I've simplified the test, re-formulated to take advantage of `bind_poly_var_top`.
I believe this not doesn't introduce any notion that wasn't already there, please take a look.

Reply:
Thanks!